src/ray/gcs/gcs_server/gcs_worker_manager.h

```diff
@@ -19,16 +19,25 @@
 #include "ray/gcs/gcs_server/usage_stats_client.h"
 #include "ray/gcs/pubsub/gcs_pub_sub.h"
 #include "ray/rpc/gcs_server/gcs_rpc_server.h"
+#include "ray/util/counter_map.h"
+#include "ray/gcs/gcs_server/gcs_init_data.h"
 
 namespace ray {
 namespace gcs {
 
+enum GcsWorkerManagerCounter {
+  kNumWorkerDeadEventsStored,
+};
+
 /// This implementation class of `WorkerInfoHandler`.
 class GcsWorkerManager : public rpc::WorkerInfoHandler {
  public:
-  explicit GcsWorkerManager(std::shared_ptr<gcs::GcsTableStorage> gcs_table_storage,
+  explicit GcsWorkerManager(size_t max_num_worker_events,
+                            std::shared_ptr<gcs::GcsTableStorage> gcs_table_storage,
                             std::shared_ptr<GcsPublisher> &gcs_publisher)
-      : gcs_table_storage_(gcs_table_storage), gcs_publisher_(gcs_publisher) {}
+      : gcs_table_storage_(gcs_table_storage),
+        gcs_publisher_(gcs_publisher),
```

Comment:
Could you please do me a favor to pass `gcs_publisher` by value? It doesn't make sense to pass by lvalue reference and make a copy.

Reply:
Sorry, this gcs_publisher is the original implementation method. I mainly added the max_num_deadw_workers variable.